### PR TITLE
add pie chart explosion opt to charts

### PR DESCRIPTION
**Pull request by sam-pitch**
Adds the "explosion" prop to PptxGenJS. This will allow us to get something approximating our pie chart padding

### CODE REVIEW

File: src/gen-charts.ts [hunk 1]
```diff
@@ -1563,6 +1563,9 @@ function makeChartType(chartType: CHART_NAME, data: IOptsChartData[], opts: ICha
 			strXml += '<c:' + chartType + 'Chart>'
 			strXml += '  <c:varyColors val="1"/>'
 			strXml += '<c:ser>'
+			if (opts.chartPieExplosion) {
```
Comment (tjinauyeung) [inline src/gen-charts.ts]: we'll need to do a couple of safety checks before passing it to the xml i.e. is it a valid number?

File: src/core-interfaces.ts [hunk 2]
```diff
@@ -1519,6 +1519,7 @@ export interface IChartPropsChartPie {
 	 * @default 0
 	 */
 	firstSliceAng?: number
+	chartPieExplosion?: number
```
Comment (sam-pitch) [inline src/core-interfaces.ts:1522]: Does this protect us from bad inputs without having to add manual inline checks? @tjinauyeung

Comment (tjinauyeung) [reply src/core-interfaces.ts:1522]: No, these definitions are not checked unfortunately. We're not using any of the TS definitions when using the library.

File: src/gen-charts.ts (hunk 1 above)
Comment (sam-pitch) [reply src/gen-charts.ts]: is the following fine?
```js
if (opts.chartPieExplosion && Number.isInteger(opts.chartPieExplosion)) {
```

Comment (tjinauyeung) [reply src/gen-charts.ts]: Do we support floats? And is there a min or max limit? I've seen some PPTX errors before because the number were out of range.

Comment (sam-pitch) [reply src/gen-charts.ts]: and Explosion needs to be an integer value according to [c-rex](https://c-rex.net/projects/samples/ooxml/e1/Part4/OOXML_P4_DOCX_explosion_topic_ID0EAIBQB.html?hl=explosion)

I don't think we should be overly defensive here. The values which Pitch can produce are quite narrow (0 - 10).

**Review by zarkone — APPROVED** [review]

**Pull request merged**